packages/sdk-middleware-http/test/http.spec.js

```diff
@@ -59,6 +59,37 @@ describe('Http', () => {
       httpMiddleware(next)(request, response)
     }))
 
+  test('should accept HEAD request and return without response body', () =>
+    new Promise((resolve, reject) => {
+      const request = createTestRequest({ uri: '/foo/foo', method: 'HEAD' })
+      const response = { resolve, reject }
+      const next = (req, res) => {
+        expect(res).toEqual({
+          ...response,
+          statusCode: 200,
+          headers: {
+            'content-type': ['application/json'],
+          },
+        })
+        resolve()
+      }
+      // Use default options
+      const httpOptions = {
+        host: testHost,
+        includeResponseHeaders: true,
+        fetch,
+      }
+      const httpMiddleware = createHttpMiddleware(httpOptions)
+      nock(testHost)
+        .defaultReplyHeaders({
+          'Content-Type': 'application/json',
+        })
+        .head('/foo/foo')
+        .reply(200, {})
```

Comment:
Maybe this might help clarify things.

You reply with an empty body `{}` here, which is not the same as when the body does not exist at all. If you change it to `.reply(200)` you should see the real issue here.